kube-derive/src/lib.rs

```diff
@@ -111,6 +111,12 @@ mod custom_resource;
 /// Adding `#[kube(derive = "PartialEq")]` is required if you want your generated
 /// top level type to be able to `#[derive(PartialEq)]`
 ///
+/// ### `#[kube(derive_schema = false)]`
+/// Disables the automatic `#[derive(JsonSchema)]` on the top level generated type.
+///
+/// This can be used to provide a completely custom schema, or to interact with third-party custom resources,
+/// where you are not responsible for installing the `CustomResourceDefinition`.
+///
```

Comment:
I would add a note that from kubernetes 1.22, the legacy apiextensions.k8s.io/v1beta1 is removed, so a schema is ultimately required. https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122

Reply:
Yeah.. working on a redesign of the attribute...